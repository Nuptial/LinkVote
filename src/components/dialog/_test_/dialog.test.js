import React from "react";
import ReactDOM from 'react-dom';
import Dialog from '../dialog';

import { render, cleanup } from '@testing-library/react';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  const selectedLink= {
    name:''
  };

  ReactDOM.render(<Dialog selectedLink={selectedLink}></Dialog>, div);
});

it('renders dialog correctly', () => {
  const {getByTestId} = render(<Dialog selectedLink={{name: 'test'}}></Dialog>)

  expect(getByTestId('dialog-span')).toHaveTextContent('test')
});

it('matches snapshot', () => {
  const tree = renderer.create(<Dialog selectedLink={{ name: "test" }}></Dialog>).toJSON();

  expect(tree).toMatchSnapshot();
});