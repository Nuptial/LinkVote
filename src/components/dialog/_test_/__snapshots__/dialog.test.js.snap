// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="dialog-container"
  >
    <div
      className="dialog-header"
    >
      Remove Link
    </div>
    <div
      className="dialog-body"
    >
      <p>
        Do you want to remove:
        <span
          data-testid="dialog-span"
        >
          test
        </span>
      </p>
    </div>
    <div
      className="dialog-footer"
    >
      <button
        className="dialog-button"
      >
        OK
      </button>
      <button
        className="dialog-button"
      >
        CANCEL
      </button>
    </div>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times fa-w-11 close-icon"
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>,
  <div
    className="overlay"
  />,
]
`;
